proxy-node-translator/src/test/java/uk/gov/ida/notification/translator/apprule/HubResponseTranslatorAppRuleTests.java

```diff
@@ -91,78 +95,66 @@ public void setup() throws Throwable {
 
-        Response eidasResponse = extractEidasResponse(buildSignedHubResponse());
+        Response eidasResponse = extractEidasResponseFromTranslator(buildSignedHubResponse());
 
         Signature signature = eidasResponse.getSignature();
 
-        assertNotNull("SAML Response needs to be signed", signature);
+        assertNotNull("EIDAS SAML Response needs to be signed", signature);
         assertTrue("Invalid signature", signatureValidator.validate(eidasResponse, null, Response.DEFAULT_ELEMENT_NAME));
         assertEquals(SignatureConstants.ALGO_ID_SIGNATURE_RSA_SHA256, signature.getSignatureAlgorithm());
     }
 
     @Test
     public void shouldReturnAnEncryptedEidasResponse() throws Exception {
-        Response eidasResponse = extractEidasResponse(buildSignedHubResponse());
+        Response eidasResponse = extractEidasResponseFromTranslator(buildSignedHubResponse());
         assertEquals(1, eidasResponse.getEncryptedAssertions().size());
         assertThat(eidasResponse.getAssertions()).isEmpty();
     }
 
     @Test
-    @Ignore
-    public void postingHubResponseShouldReturnEidasResponseForm() throws Exception {
-        Response hubResponse = buildSignedHubResponse();
-        Credential decryptingCredential = new TestCredentialFactory(TEST_PUBLIC_CERT, TEST_PRIVATE_KEY).getDecryptingCredential();
+    public void shouldDecryptAndReadEidasAssertion() throws Exception {
 
-        Response eidasResponse = extractEidasResponse(hubResponse);
-        Response decryptedEidasResponse = decryptResponse(eidasResponse, decryptingCredential);
-        Assertion eidasAssertion = decryptedEidasResponse.getAssertions().get(0);
-        Element attributeStatement = MARSHALLER.marshallToElement(eidasAssertion.getAttributeStatements().get(0));
+        Response hubResponse = buildSignedHubResponse();
+        Response eidasResponse = extractEidasResponseFromTranslator(hubResponse);
+        Response decryptedEidasResponse = decryptResponse(eidasResponse, eidasDecryptingCredential);
 
         assertEquals(hubResponse.getInResponseTo(), eidasResponse.getInResponseTo());
-        assertEquals(1, eidasAssertion.getAttributeStatements().size());
-        assertEquals(1, eidasAssertion.getAuthnStatements().size());
-
-        assertEquals("Jazzy Harold", attributeStatement.getFirstChild().getTextContent());
-    }
 
-    @Test
-    @Ignore
-    public void shouldNotAcceptUnsignedHubResponse() throws Exception {
-        javax.ws.rs.core.Response response = postHubResponseToTranslator(buildUnsignedHubResponse());
-        String message = response.readEntity(String.class);
-        assertEquals(response.getStatus(), HttpStatus.SC_BAD_REQUEST);
-        assertThat(message).contains("Error handling hub response");
+        TranslatedHubResponseTestAssertions.assertAssertionStatements(decryptedEidasResponse);
```

Comment:
It's not clear from the method name `assertAssertionStatements ` what it does. I would call it something like `checkAssertionStatementsCorrect` or `checkAssertionStatementsValid`

Reply:
done